File: go/lv0/제곱수_판별하기/제곱수_판별하기.go
import "math"

func solution(n int) int {
    // math.Sqrt는 float64만 받는다. 결과값도 그렇다.
    sqrt := math.Sqrt(float64(n))
    // sqrt를 int로 바꾸면 소수점이 잘린다. 이를 다시 float64로 바꾸면
    // n.000...0 이 된다. 만약 sqrt가 자연수라면 둘이 같을 것이다.
    if float64(int(sqrt)) == sqrt {
        return 1
    } else {
        return 2
    }
}

File: cpp/lv0/제곱수_판별하기/제곱수_판별하기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(int n) {
    int sqrt = std::sqrt(n);
    // C++에서는 다른 타입끼리 연산을 지원한다 후후
    if (static_cast<int>(sqrt) < std::sqrt(n)) {
        return 2;
    } else {
        return 1;
    }
}